test/functorch/test_vmap.py

```diff
@@ -2756,6 +2756,35 @@ def f(x):
             )(x)
             self.assertEqual(output, expected)
 
+    @parametrize('in_dim', [0, 1, 2])
+    @parametrize('out_dim', [0, 1, 2])
+    @parametrize('randomness', ['error', 'same'])
+    def test_vmap_chunksize(self, in_dim, out_dim, randomness):
+
+        x = torch.randn(4, 5, 6)
+
+        def f(x):
+            y = x.sin()
+            if randomness != "error":
+                y = y + torch.rand_like(x)
+            return y
```

Comment:
For robustness, let's test some more cases.
- inputs are some arbitrary pytree and outputs are some arbitrary pytree.
- test inputs that are not Tensors (I see there is special logic in your implementation to handle those).
- vmap(vmap (IIRC tensor_split has composite compliance issues? So just to sanity check that vmap of vmap works...)
- error cases (I'm not sure that an incompatible out_dims reports the correct error message)

Reply:
> vmap(vmap (IIRC tensor_split has composite compliance issues? So just to sanity check that vmap of vmap works...)

We don't use the overload of `tensor_split` which has composite compliance issue.